pypose/optim/kernel.py

```diff
@@ -245,18 +245,17 @@ def __init__(self, a: float = 1.0, b: float = -1.0) -> None:
         super().__init__()
         assert a > 0, ValueError("a has to be positive: {}".format(a))
         assert b < 0, ValueError("b has to be negative: {}".format(b))
-        self.a = a
-        self.b = b
+        self.a, self.b = a, b
 
     def forward(self, input: Tensor) -> Tensor:
         '''
         Args:
             input (torch.Tensor): the input tensor (non-negative).
         '''
         assert torch.all(input >= 0), 'input has to be non-negative'
-        part1 = (1 + ((input-self.a) / self.b).exp()).log()
-        part2 = (1 + (-self.a / self.b).exp())
-        return self.b * part1 - self.b * part2
+        res = self.b * (1 + ((input-self.a) / self.b).exp()).log()
+        offset = self.b * math.log((1 + math.exp(-self.a / self.b)))
+        return res - offset
```

Comment:
How about
```
result = self.b * (1 + ((input-self.a) / self.b).exp()).log()
offset = self.b * math.log((1 + math.exp(-self.a / self.b))) # constant, no grad.
return result - offset
```

Reply:
You mean replacing `res` with `result`?

Reply:
yes, this will make the two variables have same name length.
And add the short comments to explain why we use `math` library not `torch` library.